src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/updaters/PtvCodeListEntityUpdater.java

```diff
@@ -0,0 +1,251 @@
+        orderIndex++;
+      }
+    } else {
+      logger.log(Level.SEVERE, () -> String.format("Failed to update code list [%d]: %s", response.getStatus(), response.getMessage()));
+    }
+  }
+
+
+  private void handleCodeListPagedResponse(ApiResponse<VmOpenApiCodeListPage> response, CodeType codeType) {
+    Long orderIndex = codeType.ordinal() * 100000l;
+    
+    if (response.isOk()) {
+      for (VmOpenApiCodeListItem item : response.getResponse().getItemList()) {
+        handleItem(orderIndex, item, codeType);
+        orderIndex++;
+      }
+    } else {
+      logger.log(Level.SEVERE, () -> String.format("Failed to update paged code list [%d]: %s", response.getStatus(), response.getMessage()));
+    }
+  }
+
+  private void handleCountryCodeListResponse(ApiResponse<List<VmOpenApiDialCodeListItem>> response, CodeType codeType) {
+    Long orderIndex = codeType.ordinal() * 100000l;
+    if (response.isOk()) {
+      for (VmOpenApiDialCodeListItem item : response.getResponse()) {
+        handleDialCodeItem(orderIndex, item, codeType);
+        orderIndex++;
+      }
+    } else {
+      logger.log(Level.SEVERE, () -> String.format("Failed to update country code list [%d]: %s", response.getStatus(), response.getMessage()));
+    }
+  }
+  
+  private void handleDialCodeItem(Long orderIndex, VmOpenApiDialCodeListItem ptvItem, CodeType codeType) {
+    CodeId ptvCodeId = ptvIdFactory.createCodeId(String.format("%s-%s", codeType, ptvItem.getCode()));
+    Identifier identifier = identifierController.acquireIdentifier(orderIndex, ptvCodeId);
+    CodeId kuntaApiCodeId = kuntaApiIdFactory.createFromIdentifier(CodeId.class, identifier);
+    updateItem(orderIndex, kuntaApiCodeId, ptvTranslator.translateCode(kuntaApiCodeId, ptvItem, codeType));
+  }
+
+  private void handleItem(Long orderIndex, VmOpenApiCodeListItem ptvItem, CodeType codeType) {
+    CodeId ptvCodeId = ptvIdFactory.createCodeId(String.format("%s-%s", codeType, ptvItem.getCode()));
+    Identifier identifier = identifierController.acquireIdentifier(orderIndex, ptvCodeId);
+    CodeId kuntaApiCodeId = kuntaApiIdFactory.createFromIdentifier(CodeId.class, identifier);
+    updateItem(orderIndex, kuntaApiCodeId, ptvTranslator.translateCode(kuntaApiCodeId, ptvItem, codeType));
+  }
+  
+  private void updateItem(Long orderIndex, CodeId kuntaApiCodeId, Code code) {
+    if (code != null) {
+      modificationHashCache.put(kuntaApiCodeId.getId(), createPojoHash(code));
+      ptvCodeResourceContainer.put(kuntaApiCodeId, code);
+      index(kuntaApiCodeId.getId(), code, orderIndex);
+    } else {
+      logger.log(Level.SEVERE, () -> String.format("Failed to translate code %s", kuntaApiCodeId));
+    }
+  }
+  
+  private void index(String kuntaApiId, Code code, Long orderIndex) {
+    String nameFi = null;
+    String nameSv = null;
+    String nameEn = null;
+    
+    for (LocalizedValue localizedValue : code.getNames()) {
+      switch (localizedValue.getLanguage()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)